src/org/traccar/reports/ReportUtils.java

```diff
@@ -35,4 +39,46 @@ private ReportUtils() {
         return result;
     }
 
+    public static double calculateDistance(Position firstPosition, Position lastPosition) {
+        return calculateDistance(firstPosition, lastPosition, true);
+    }
+
+    public static double calculateDistance(Position firstPosition, Position lastPosition, boolean useOdometer) {
+        BigDecimal distance = new BigDecimal("0.0");
```

Comment:
I get PMD complain "Avoid creating BigDecimal with a decimal (float/double) literal. Use a String literal". Int is OK.

As I understand using `BigDecimal.setScale` is best (precise) way to round floating point values.
And really sometimes I got *999999999 values subtracting two rounded distances.
I used double, and rounded it at the end of function, but got PMD warning, I converted double to string and ... saw that there is casting from odometer to double and from distance to double. I Thought why not to cast directly to BigDecimal and do math with it.